priam/src/main/java/com/netflix/priam/aws/SDBInstanceData.java

```diff
@@ -155,13 +180,17 @@ public void createInstance(PriamInstance instance) throws AmazonServiceException
      *
      * @throws AmazonServiceException
      */
-    public void registerInstance(PriamInstance instance) throws AmazonServiceException {
+    public void registerInstance(PriamInstance instance, String expectedPreviousInstanceId) throws AmazonServiceException {
         logger.info("Registering PriamInstance in SimpleDB: {}", instance);
         AmazonSimpleDB simpleDBClient = getSimpleDBClient();
         PutAttributesRequest putReq = new PutAttributesRequest(sdbDomain, getKey(instance), createAttributesToRegister(instance));
         UpdateCondition expected = new UpdateCondition();
         expected.setName(Attributes.INSTANCE_ID);
-        expected.setExists(false);
+        if (expectedPreviousInstanceId == null) {
+            expected.setExists(false);
+        } else {
+            expected.setValue(expectedPreviousInstanceId);
```

Comment:
Should there be a following statement after this line:
`expected.setExists(true)`
This would make sure that the update only goes through if the expected instance value exists.